poppy/instrument.py

```diff
@@ -852,27 +855,30 @@ def _get_weights(self, source=None, nlambda=None, monochromatic=None, verbose=Fa
         """ Return the set of discrete wavelengths, and weights for each wavelength,
         that should be used for a PSF calculation.
 
-        Uses pysynphot (if installed), otherwise assumes simple-minded flat spectrum
+        Uses stsynphot (if installed), otherwise assumes simple-minded flat spectrum
 
         """
         if nlambda is None or nlambda == 0:
             nlambda = self._get_default_nlambda(self.filter)
 
         if monochromatic is not None:
             poppy_core._log.info("Monochromatic calculation requested.")
-            return (np.asarray([monochromatic]), np.asarray([1]))
+            return ([monochromatic] * units.AA, np.asarray([1]))
```

Comment:
Ops, I tried something but forgot to revert.

```suggestion
            return (np.asarray([monochromatic]), np.asarray([1]))
```